velox/common/caching/SsdFile.cpp

```diff
@@ -638,57 +638,40 @@ void SsdFile::deleteCheckpoint(bool keepLog) {
   }
 
   if (evictLogWriteFile_ != nullptr) {
-    try {
-      if (keepLog) {
-        truncateEvictLogFile();
-      } else {
-        deleteEvictLogFile();
-      }
-    } catch (const std::exception& e) {
-      ++stats_.deleteCheckpointErrors;
-      VELOX_SSD_CACHE_LOG(ERROR) << "Error in deleting evictLog: " << e.what();
+    if (keepLog) {
+      truncateFile(evictLogWriteFile_.get());
+    } else {
+      auto logPath = getEvictLogFilePath();
+      deleteFile(std::move(evictLogWriteFile_), logPath);
     }
   }
 
   if (checkpointWriteFile_ != nullptr) {
-    deleteCheckpointFile();
+    auto checkpointPath = getCheckpointFilePath();
+    deleteFile(std::move(checkpointWriteFile_), checkpointPath);
   }
 }
 
-void SsdFile::truncateEvictLogFile() {
-  VELOX_CHECK_NOT_NULL(evictLogWriteFile_);
-  evictLogWriteFile_->truncate(0);
-  evictLogWriteFile_->flush();
-}
-
-void SsdFile::truncateCheckpointFile() {
-  VELOX_CHECK_NOT_NULL(checkpointWriteFile_);
-  checkpointWriteFile_->truncate(0);
-  checkpointWriteFile_->flush();
+void SsdFile::truncateFile(WriteFile* file) {
+  VELOX_CHECK_NOT_NULL(file);
+  file->truncate(0);
+  file->flush();
 }
 
-void SsdFile::deleteEvictLogFile() {
-  VELOX_CHECK_NOT_NULL(evictLogWriteFile_);
-  evictLogWriteFile_->close();
-  evictLogWriteFile_.reset();
-  const auto evictLogFilePath = getEvictLogFilePath();
-  if (fs_->exists(evictLogFilePath)) {
-    fs_->remove(evictLogFilePath);
-  }
-}
-
-void SsdFile::deleteCheckpointFile() {
-  VELOX_CHECK_NOT_NULL(checkpointWriteFile_);
+void SsdFile::deleteFile(
+    std::unique_ptr<WriteFile> file,
+    std::string& filePath) {
+  VELOX_CHECK_NOT_NULL(file);
   try {
-    checkpointWriteFile_->close();
-    checkpointWriteFile_.reset();
-    const auto checkpointFilePath = getCheckpointFilePath();
-    if (fs_->exists(checkpointFilePath)) {
-      fs_->remove(checkpointFilePath);
+    file->close();
+    file.reset();
+    if (fs_->exists(filePath)) {
+      fs_->remove(filePath);
     }
   } catch (const std::exception& e) {
-    ++stats_.deleteCheckpointErrors;
-    VELOX_SSD_CACHE_LOG(ERROR) << "Error in deleting checkpoint: " << e.what();
+    ++stats_.deleteFileErrors;
+    VELOX_SSD_CACHE_LOG(ERROR)
+        << fmt::format("Error in deleting file {}: {}", filePath, e.what());
```

Comment:
we don't need file path as file object provides name method? Thanks!

Reply:
file object doesn't provide name method. shall we add one?

Reply:
It has getName() and can you rename to read() if it doesn't break Prestissimo?